addons/mail/models/mail_template.py

```diff
@@ -351,7 +380,10 @@ def render_template(self, template_txt, model, res_ids, post_process=False):
             template = mako_env.from_string(tools.ustr(template_txt))
         except Exception:
             _logger.info("Failed to load template %r", template_txt, exc_info=True)
-            return multi_mode and results or results[res_ids[0]]
+            raise UserError(_(
```

Comment:
I think I already asked it but why raising here ? In several places we expect to receive a void string if it crashes. I think changing it implies checking all calls to render_template. It seems to be bug prone to me. What do you think ?

Reply:
@tde-banana-odoo If the user sends the template on a wizard, without saving, the rendering might still crash but the mail will be sent anyways (empty mail).

I was actually thinking of raising the exception in the write override since that's all that's needed to block it when saving, but thought maybe we wanted the same for when sending directly without saving a template.

Copy-pasted, but to add to that, yes you're right it's error-prone, I guess we can assume users won't be sending fancy emails through the wizard :^)